### PR TITLE
WWST-6945 - support for Somfy Situo 4 Zigbee Pure

### PR DESCRIPTION
**Pull request by KKlimczukS**
WWST-6945 - support for Somfy Situo 4 Zigbee Pure

@greens @dkirker @tpmanley 
Device supports Power Configuration cluster on endpoint 0xE8
Instead of OnOff cluster - Window Covering cluster is supported (for Somfy Situo 4) - because there is a middle ("My") button that sends "stop" commands.

Please, could You take a look at the code ?
cc: @SmartThingsCommunity/srpol-pe-team

### CODE REVIEW

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy [hunk 1]
```diff
@@ -170,11 +190,19 @@ def updated() {
...
 
-	def cmds = zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, 0x21, DataType.UINT8, 30, 21600, 0x01) +
-			zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x21) +
-			zigbee.addBinding(zigbee.ONOFF_CLUSTER) +
-			readDeviceBindingTable() // Need to read the binding table to see what group it's using
+	def cmds = []
+
+	if (isSomfySituo()) {
+		cmds += zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, 0x21, DataType.UINT8, 30, 21600, 0x01, ["destEndpoint":0xE8]) +
+				zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x21, ["destEndpoint":0xE8]) +
```
Comment (tpmanley) [inline devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: I'd put the reads before the configureReporting so we get the current state more quickly

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy [hunk 2]
```diff
@@ -188,7 +216,14 @@ def parse(String description) {
 		if ((description?.startsWith("catchall:")) || (description?.startsWith("read attr -"))) {
 			def descMap = zigbee.parseDescriptionAsMap(description)
 			if (descMap.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.attrInt == 0x0021) {
-				event = getBatteryEvent(zigbee.convertHexToInt(descMap.value))
+				def batteryValue
+				if (isSomfySituo()) {
+					// Somfy Situo reports decimal values
+					batteryValue = descMap.value
```
Comment (tpmanley) [inline devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: This sounds like a device. Has this been discussed with Somfy?

Comment (greens) [reply devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: @tpmanley did you mean "device-specific behavior?" or something?

Comment (tpmanley) [reply devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: Oops, I meant to say "device bug"

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy (hunk 1 above)
Comment (KKlimczukS) [reply devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: ok, done.

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy (hunk 2 above)
Comment (KKlimczukS) [reply devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: @greens @tpmanley 
nah, the device reports hex values, i inserted new batteries to verify that (previously i was witnessing only small values from range (0-99).

File: devicetypes/smartthings/ikea-button.src/ikea-button.groovy [hunk 3]
```diff
@@ -188,7 +216,12 @@ def parse(String description) {
 		if ((description?.startsWith("catchall:")) || (description?.startsWith("read attr -"))) {
 			def descMap = zigbee.parseDescriptionAsMap(description)
 			if (descMap.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.attrInt == 0x0021) {
-				event = getBatteryEvent(zigbee.convertHexToInt(descMap.value))
+				def batteryValue = zigbee.convertHexToInt(descMap.value)
+				if (isSomfySituo()) {
```
Comment (tpmanley) [inline devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: Shouldn't need a manufacturer check here. This attribute reports the percentage in .5% increments so it should always be divided by 2. If there are devices that use this DTH that report the value incorrectly according to the spec then the manufacturer/model check should be for those devices.

Comment (KKlimczukS) [reply devicetypes/smartthings/ikea-button.src/ikea-button.groovy]: ok, since this dth only supported ikea devices (so far) - i reversed that condition.

**Review by greens — APPROVED** [review]
lgtm, but will let @tpmanley give his approval before merging

nvm, just merging

**Pull request merged**